codec-http2/src/main/java/io/netty/handler/codec/http2/Http2MultiplexCodec.java

```diff
@@ -1197,18 +1185,16 @@ private ChannelFuture write0(Object msg) {
 
             @Override
             public void flush() {
-                if (!writeDoneAndNoFlush) {
+                // If we are currently in the parent channel's read loop we should just ignore the flush.
+                // We will ensure we trigger ctx.flush() after we processed all Channels later on and
+                // so aggregate the flushes. This is done as ctx.flush() is expensive when as it may trigger an
+                // write(...) or writev(...) operation on the socket.
+                if (!writeDoneAndNoFlush || parentReadInProgress) {
```

Comment:
@Scottmitch I wonder if we should always flush if `channel.isWritable()` returns false here to try to make room again... Something like:

```java 
if (!writeDoneAndNoFlush || (parentReadInProgress && channel.isWritable()) {
}
```

Reply:
Perhaps, but I think we need a more comprehensive review of writability for the child channel implementation is warranted https://github.com/netty/netty/issues/8148.

For example the child channel writes will not actually be written to the parent channel until the parent channel's `flush()` is called. This is when flow control is done. So I wouldn't expect `childChannel.parent().isWritable()` to be impacted by this operation.

Reply:
ok thanks :)